### PR TITLE
fix bit aggregation function bug

### PR DESCRIPTION
**Pull request by qingxinhome**
## What type of PR is this?

- [ ] API-change
- [x] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #3734

## What this PR does / why we need it:
fix bit aggregation function bug which bit_wise function result different in ARM and X86

### CODE REVIEW

**Review by fengttt — COMMENTED** [review]
can someone explain why we need to have bit op on floating point types?

**Review by heni02 — APPROVED** [review]

**Comment by qingxinhome** [comment]
> can someone explain why we need to have bit op on floating point types?

This is for compatibility with MySQL semantics. @fengttt

File: pkg/sql/colexec/agg/bit_and.go [hunk 1]
```diff
@@ -59,6 +62,9 @@ func (ba *BitAnd[T1]) Fill(groupIndex int64, v1 T1, v2 uint64, z int64, isEmpty
 	if isEmpty {
 		v2 = ^uint64(0)
 	}
+	if float64(v1) > math.MaxUint64 {
+		return math.MaxInt64 & v2, false
+	}
 	return uint64(v1) & v2, false
```
Comment (nnsgmsone) [inline pkg/sql/colexec/agg/bit_and.go:68]: Is this the semantics of mysql. What is the meaning of this a..

**Review by aressu1985 — APPROVED** [review]

File: pkg/sql/colexec/agg/bit_and.go (hunk 1 above)
Comment (qingxinhome) [reply pkg/sql/colexec/agg/bit_and.go:68]: When x86 and M1 overflow the result of uint64 type conversion, the processing methods are different, which leads to the generation of bugs. Therefore, it is used to directly determine whether the uint64 range is exceeded. If the uint64 range is exceeded, it is calculated according to the maximum value of int64, which is consistent with MySQL

**Review by nnsgmsone — APPROVED** [review]

**Pull request merged**